File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
  let result = 0
  let prevArr = arr
  while(true) {
      const newArr = prevArr.map(x => {
          if(x >= 50 && !(x % 2)) return x/2
          if(x < 50 && x % 2) return x*2+1
          return x
      }) 
      if(JSON.stringify(prevArr) === JSON.stringify(newArr)) break
      result += 1
      
      prevArr = newArr
  }
  return result
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.Arrays;

public class Solution {
    public int solution(int[] arr) {
        int count = 0;
        int[] prevArr;
        
        while (true) {
            prevArr = arr.clone();
            
            for (int i = 0; i < arr.length; i++) {
                if (arr[i] >= 50 && arr[i] % 2 == 0) {
                    arr[i] /= 2;
                } else if (arr[i] < 50 && arr[i] % 2 != 0) {
                    arr[i] = arr[i] * 2 + 1;
                }
            }
            
            count++; 
            if (Arrays.equals(arr, prevArr)) {
                count--;
                break;
            }
        }
        
        return count;
    }
}
